src/tdastro/astro_utils/passbands.py

```diff
@@ -56,44 +74,92 @@ def __str__(self) -> str:
+        self, delta_wave: Optional[float] = 5.0, trim_quantile: Optional[float] = 1e-3
+    ):
+        """Process the transmission tables for all passbands in the group; recalculate group's wave attribute.
+
+        Parameters
+        ----------
+        delta_wave : float or None, optional
+            The grid step of the wave grid. Default is 5.0.
+        trim_quantile : float or None, optional
+            The quantile to trim the transmission table by. For example, if trim_quantile is 1e-3, the
+            transmission table will be trimmed to include only the central 99.8% of rows.
+        """
+        for passband in self.passbands.values():
+            passband.process_transmission_table(delta_wave, trim_quantile)
+
+        self._update_waves()
+
+    def fluxes_to_bandfluxes(self, flux_density_matrix: np.ndarray) -> np.ndarray:
         """Calculate bandfluxes for all passbands in the group.
 
         Parameters
         ----------
         flux_density_matrix : np.ndarray
             A 2D array of flux densities where rows are times and columns are wavelengths.
-        wavelengths_angstrom : np.ndarray
-            An array of wavelengths in Angstroms corresponding to the columns of flux_density_matrix.
 
         Returns
         -------
-        dict
-            A dictionary of bandfluxes with passband labels as keys.
+        dict of np.ndarray
+            A dictionary of bandfluxes with passband names as keys and np.ndarrays of bandfluxes as values.
         """
+        if flux_density_matrix.size == 0 or len(self.waves) != len(flux_density_matrix[0]):
+            flux_density_matrix_num_cols = 0 if flux_density_matrix.size == 0 else len(flux_density_matrix[0])
+            raise ValueError(
+                f"PassbandGroup mismatched grids: Flux density matrix has {flux_density_matrix_num_cols} "
+                f"columns, which does not match transmission table's {len(self.waves)} rows. Check that the "
+                f"flux density matrix was calculated on the same grid as the transmission tables, which can "
+                f"be accessed via the Passband's or PassbandGroup's waves attribute."
+            )
+
         bandfluxes = {}
-        for label, passband in self.passbands.items():
-            bandfluxes[label] = passband.fluxes_to_bandflux(flux_density_matrix, wavelengths_angstrom)
+        for full_name, passband in self.passbands.items():
+            # We only want the fluxes that are in the passband's wavelength range
+            # So, find the indices in the group's wave grid that are in the passband's wave grid
+            lower, upper = passband.waves[0], passband.waves[-1]
+            lower_index, upper_index = np.searchsorted(self.waves, [lower, upper])
+            # Check that this is the right grid after all (check will fail if the grid is not regular, or
+            # passbands are overlapping)
+            if np.array_equal(self.waves[lower_index : upper_index + 1], passband.waves):
```

Comment:
Again, I think this could be precomputed during data creation and stored in a dictionary mapping full_name to a boolean indicating that the arrays are equal.

Reply:
Quoted in #112